src/htmx.js

```diff
@@ -1849,17 +1845,23 @@ return (function () {
 
         function findElementsToProcess(elt) {
             if (elt.querySelectorAll) {
-                var boostedElts = hasChanceOfBeingBoosted() ? ", a, form" : "";
-                var results = elt.querySelectorAll(VERB_SELECTOR + boostedElts + ", [hx-sse], [data-hx-sse], [hx-ws]," +
+                var results = elt.querySelectorAll(VERB_SELECTOR + ", a, form, [type='submit']" + ", [hx-sse], [data-hx-sse], [hx-ws]," +
```

Comment:
To avoid processing each `a` and `form` element without a `hx-boost` attribute, I would suggest changing the selector from `a, form` to
```js 
a[hx-boost], a[data-hx-boost], [hx-boost] a, [data-hx-boost] a,
form[hx-boost], form[data-hx-boost], [hx-boost] form, [data-hx-boost] form
```

The old implementation already did this, but not really strictly.
And there is no need to concat the strings `..[type='submit']" + ", [hx-sse]...`.

Reply:
We actually want htmx to process non-boosted `a` and `form` elements here since some extensions can have behaviors depending on them, that are outside htmx's initial scope.
For example, one of the linked issues was related to `ws-send` that can be placed on a child form, which shouldn't require to be boosted, see [the websocket extensions tests](https://github.com/bigskysoftware/htmx/pull/1559/files#diff-5d44e06c69b1e00b7645e380407a00efeb408254485e448e6262aa4237b69445R437)

Htmx processing non-htmx elements will, in the worst case, simply result in those elements having an empty internal data for links, and the `lastButtonClicked` property in internal data for forms (since that element won't declare hx-on, hx-trigger & so on, htmx won't process anything on it)

Reply:
You're right about the useless concatenation though, let me fix that

Reply:
>We actually want htmx to process non-boosted a and form elements here since some extensions can have...

Then we could also add `[hx-ext] a, [hx-ext] form` to solve these problems, couldn`t we?

>Htmx processing non-htmx elements will, in the worst case, simply result in those elements having an empty

Apart from what you said, htmx will also trigger events, register event listeners (initButtonTracking), but most of the time you do not use `hx-boost` or an extension.

Reply:
I'm not familiar with the extensions system, will an element always be a descendant of a `hx-ext` element ?

Reply:
I would say yes, as described in the documentation:
>You will need to include the javascript file for the extension and then install it using the `hx-ext` attributes.

>For example, one of the linked issues was related to ws-send that can be placed on a child form, which shouldn't require to be boosted, see [the websocket extensions tests](https://github.com/bigskysoftware/htmx/pull/1559/files#diff-5d44e06c69b1e00b7645e380407a00efeb408254485e448e6262aa4237b69445R437)

I had a look at the websocket extension code and IMO an extension needs to take care of handling special attributes like `ws-send` itself. So I am not sure why we should initialize links or forms without `hx-boost` at all.

BTW: I adjust my code to check if all my tests are green, even without the additional selector `[hx-ext] a, [hx-ext] form`. And yep, all tests are fine.

Reply:
After reading the issue https://github.com/bigskysoftware/htmx/issues/1337 I'm sure we need the additional selector `[hx-ext] form` because of the button tracking stuff for form elements.

Reply:
Ah and don't forget `[data-hx-ext] form`.

Reply:
Now that I think about it, I think there's another issue with this selectors set.
Htmx runs them relatively to the root element that's being initialized.
Which means, if htmx were to process a `form` that's dynamically appended to a `hx-ext` ascendant (whether it be through standard htmx attributes, or manually + a `htmx.process` call), that element would be ignored, since it wouldn't match the selector

Something like
```html
<div hx-ext="whatever">
    <div id="target"></div>
</div>

<button hx-post="/url" hx-target="#target">Click me</button>
```
If `/url` responds with a form, that gets swapped inside target, with your suggested selectors, that form won't be processed since it wouldn't match any of those selectors (as it'd be run on the `target` element, not the document root)

I'll probably add some tests to ensure htmx processes correctly dynamically added elements as well

Reply:
I was skeptical at first about the real gain of this, given that it adds "a lot" (subjective ofc) of code just to avoid processing elements and firing 2 events, that would ultimately cause no issue anyway (since htmx doesn't get in the way of standard / expected behaviours for those `a` and `form` elements),

Decided to benchmark it with [this JSFiddle](https://jsfiddle.net/26kshucz/). I was honestly even expecting your code to be slightly slower
```
Finding out if form should be processed : 30 ms
Firing events : 140 ms
```
_(10 000 iterations, average results)_

Turns out firing events is actually freaking slow in comparison.

So now I'm fully convinced 😄 thanks for your help! I'll add those changes

Reply:
Pushed, all tests pass, great! Thanks again, closing this conversation